### PR TITLE
Listing secured by sensitivity of associated project and project-scoped roles

### PR DESCRIPTION
**Pull request by elisabeth-sorrell**
Components refactored:
- project
- partnership
- budget/budgetRecords
- project members

### CODE REVIEW

File: src/components/partner/partner.repository.ts [hunk 1]
```diff
@@ -106,8 +106,8 @@ export class PartnerRepository extends DtoRepository(Partner) {
           sub
             .with('projList')
             .raw('UNWIND projList as project')
-            // TODO: have this match from line 92 so we don't have to match this twice
-            .match([
+            // TODO: I don't think this is needed.... more thought on this later.
+            .optionalMatch([
```
Comment (CarsonF) [inline src/components/partner/partner.repository.ts:110]: Shouldn't `matchProjectScopedRoles` & L103 be moved into this subquery after deduping projects matched?

Comment (elisabeth-sorrell) [reply src/components/partner/partner.repository.ts]: I don't think it should. if `matchProjecScopedRoles` andL103 were moved into this subquery, we would have to double-collect the scoped roles in order to collect all scopedRoles of all projects that member is connected to. 

.... fixed the match with partnership. Fixing org now.

Comment (elisabeth-sorrell) [reply src/components/partner/partner.repository.ts]: org fixed

**Review by CarsonF — APPROVED** [review]
Looks great @elisabeth-sorrell, thanks! 🎉

**Pull request merged**